src/Xamarin.Android.Build.Tasks/Tasks/GetAdditionalResourcesFromAssemblies.cs

```diff
@@ -214,29 +280,26 @@ string MakeSureLibraryIsInPlace (string destinationBase, string url, string vers
 				Directory.CreateDirectory (zipDir);
 
 			string file = Path.Combine (zipDir, !uri.IsFile ? hash + ".zip" : Path.GetFileName (uri.AbsolutePath));
-			if (!File.Exists (file) || !IsValidDownload (file, sha1)) {
+			if (!File.Exists (file) || !IsValidDownload (file, sha1) || !MonoAndroidHelper.IsValidZip (file)) {
 				int progress = -1;
-				DownloadProgressChangedEventHandler downloadHandler = (o, e) => {
-					if (e.ProgressPercentage % 10 != 0 || progress == e.ProgressPercentage)
+				var downloadHandler = new Action<long, long, int>((r,t,p) => {
+					if (p % 10 != 0 || progress == p)
 						return;
-					progress = e.ProgressPercentage;
-					LogMessage ("\t({0}/{1}b), total {2:F1}%", e.BytesReceived,
-						e.TotalBytesToReceive, e.ProgressPercentage);
-				};
-				using (var client = new System.Net.WebClient ()) {
-					client.DownloadProgressChanged += downloadHandler;
-					LogMessage ("  Downloading {0} into {1}", url, zipDir);
-					try {
-						client.DownloadFileTaskAsync (url, file).Wait (Token);
+					progress = p;
+					LogMessage ("\t({0}/{1}b), total {2:F1}%", r,
+						t, p);
+				});
+				LogMessage ("  Downloading {0} into {1}", url, zipDir);
+				try {
+					Download (file, uri, downloadHandler);
+					if (MonoAndroidHelper.IsValidZip (file))
 						LogMessage ("  Downloading Complete");
-					} catch (Exception e) {
-						LogCodedError ("XA5208", "Download failed. Please download {0} and put it to the {1} directory.", url, destinationDir);
-						LogCodedError ("XA5208", "Reason: {0}", e.GetBaseException ().Message);
-						Log.LogMessage (MessageImportance.Low, e.ToString ());
-						if (File.Exists (file))
-							File.Delete (file);
-					}
-					client.DownloadProgressChanged -= downloadHandler;
+					else 
+						LogCodedError ("XA5208", "Download succeeded but the zip file was not valid. Please do a clean build and try again.");
+				} catch (Exception e) {
+					LogCodedError ("XA5208", "Download failed. Please do a clean build and try again.");
```

Comment:
How will "do[ing] a clean build" fix the problem, given that .zip download is usually to a location *outside* the project directory?